pagerduty/resource_pagerduty_event_orchestration_integration.go

```diff
@@ -0,0 +1,309 @@
+		Importer: &schema.ResourceImporter{
+			StateContext: resourcePagerDutyEventOrchestrationIntegrationImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"event_orchestration": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: addIntegrationMigrationWarning(),
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"label": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"parameters": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"routing_key": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func addIntegrationMigrationWarning() schema.SchemaValidateDiagFunc {
+	return func(v interface{}, p cty.Path) diag.Diagnostics {
+		var diags diag.Diagnostics
+
+		diags = append(diags, diag.Diagnostic{
+			Severity:      diag.Warning,
+			Summary:       "Modifying the event_orchestration property of the 'pagerduty_event_orchestration_integration' resource will cause all future events sent with this integration's routing key to be evaluated against the new Event Orchestration.",
+			AttributePath: p,
+		})
+
+		return diags
+	}
+}
+
+func getEventOrchestrationIntegrationPayloadData(d *schema.ResourceData) (string, *pagerduty.EventOrchestrationIntegration) {
+	orchestrationId := d.Get("event_orchestration").(string)
+
+	integration := &pagerduty.EventOrchestrationIntegration{
+		Label: d.Get("label").(string),
+	}
+
+	return orchestrationId, integration
+}
+
+func resourcePagerDutyEventOrchestrationIntegrationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := meta.(*Config).Client()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	oid, payload := getEventOrchestrationIntegrationPayloadData(d)
+
+	retryErr := resource.RetryContext(ctx, 2*time.Minute, func() *resource.RetryError {
+		log.Printf("[INFO] Creating Integration '%s' for PagerDuty Event Orchestration '%s'", payload.Label, oid)
+
+		if integration, _, err := client.EventOrchestrationIntegrations.CreateContext(ctx, oid, payload); err != nil {
+			if isErrCode(err, 400) {
+				return resource.NonRetryableError(err)
+			}
+			return resource.RetryableError(err)
+		} else if integration != nil {
+			// Try reading an integration after creation, retry if not found:
+			if _, _, readErr := client.EventOrchestrationIntegrations.GetContext(ctx, oid, integration.ID); readErr != nil {
+				log.Printf("[WARN] Cannot locate Integration '%s' on PagerDuty Event Orchestration '%s'. Retrying creation...", integration.ID, oid)
+				return resource.RetryableError(readErr)
+			} else {
+				d.SetId(integration.ID)
+				setEventOrchestrationIntegrationProps(d, integration)
+			}
```

Comment:
Concurrent requests to the same orchestration cause a race condition, that's why here and below we make another request to confirm that an integration was indeed created/updated/deleted, and retry if it wasn't. We have similar logic for event rules.